File: app/src/main/java/com/datagene/daggerexample22/di/main/ProfileFragmentBuildersModule.java
package com.datagene.daggerexample22.di.main;


import com.datagene.daggerexample22.ui.fragment.ProfileFragment;

import dagger.Module;
import dagger.android.ContributesAndroidInjector;

@Module
public abstract class ProfileFragmentBuildersModule {
    @ContributesAndroidInjector
    abstract ProfileFragment contributeProfileFragment();
}

File: app/src/main/java/com/datagene/daggerexample22/di/main/ProfileFragmentBuildersModule.kt
package com.datagene.daggerexample22.di.main

import com.datagene.daggerexample22.ui.fragment.ProfileFragment
import dagger.Module
import dagger.android.ContributesAndroidInjector

@Module
abstract class ProfileFragmentBuildersModule {
    @ContributesAndroidInjector
    abstract fun contributeProfileFragment(): ProfileFragment
}